### PR TITLE
Redux-thunk-API

### PR DESCRIPTION
**Pull request by ONJoseph**
- [x] Add redux-thunk to your project.
- [x] Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) for making API calls.
- [x] Fetch book list from Bookstore API on Books page load.
- [x] Add an Action Creator, that returns a function.
- [x] Fetch list of books.
- [x] Dispatch another action (add an action type) that will pass API response to reducer and updates the state.
- [x] Refactor your add book and remove book features to persist your changes in the server.
- [x] Change your existing Action Creators, so they return functions by using API.
- [x] Dispatch an action (action type that existed before refactor) that will update the state accordingly.
- [x] Add book and remove book should work in the same way after the refactor

### CODE REVIEW

File: src/services/bookstoreService.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import http from '../API-commons';
+
+import {
+  appsEndpoint, appIdEndpoint, itemIdEndpoint, booksEndpoint,
+} from '../modules/endpoints';
+
+const createApp = () => http.post(appsEndpoint);
+
+const createBook = (id) => http.post(appsEndpoint + appIdEndpoint(id) + booksEndpoint);
+
```
Comment (Gambit142) [inline src/services/bookstoreService.js]: - [ ] After creating the function to post to the API endpoint, you are to create an asynchronous action creator that returns a function and adds books through the API endpoint. The action creator will return a function that takes `dispatch` as a parameter. The function being returned will then dispatch the action `creator` addBook in your redux/books/books.js file. This asynchronous action creator should be dispatched in the `Form` component when a book is added. You can read this [article](https://www.bezkoder.com/react-redux-crud-example/) to have a clearer idea of what I am talking about.

#### Pay special attention to this part to have an idea of how to create the asynchronous action creator for creating a book using an API endpoint. 👍 

image

File: src/services/bookstoreService.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const getBooks = (id) => http.get(appsEndpoint + appIdEndpoint(id) + booksEndpoint);
+
```
Comment (Gambit142) [inline src/services/bookstoreService.js]: - [ ] After creating the function to retrieve data from the API endpoint, you are to create an asynchronous action creator that returns a function and fetches the books list from the API endpoint. The action creator will return a function that takes `dispatch` as a parameter. The function being returned will then dispatch a new action type that would pass the books list to the reducer and update the state (This means you will create a new `action`, `action type` and add a switch case in the `reducer` in your redux/books/books.js file). This asynchronous action creator should be dispatched in your Books component on page load (I would advise you to use the `useEfect` to dispatch the action creator for fetching data from the API). You can read this [article](https://www.bezkoder.com/react-redux-crud-example/) to have a clearer idea of what I am talking about.

#### Pay special attention to this part to have an idea of how to create the asynchronous action creator for retrieving data from an API. 👍  

image

File: src/services/bookstoreService.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const deleteBook = (appId, bookId) => http.delete(
+  appsEndpoint + appIdEndpoint(appId) + booksEndpoint + itemIdEndpoint(bookId),
+);
+
```
Comment (Gambit142) [inline src/services/bookstoreService.js]: - [ ] After creating the function to delete a single book from the API endpoint, you are to create an asynchronous action creator that returns a function and deletes a book from the API endpoint. The action creator will return a function that takes `dispatch` as a parameter. The function being returned will then dispatch the removeBook action type in your redux/books/books.js file. This asynchronous action creator should be dispatched in your Book component when you want to delete a single book from the list. You can read this [article](https://www.bezkoder.com/react-redux-crud-example/) to have a clearer idea of what I am talking about.

#### Pay special attention to this part to have an idea of how to create the asynchronous action creator for deleting a book. 👍  

image

File: src/redux/configureStore.js [hunk 4]
```diff
@@ -1,5 +1,10 @@
 // Bookstore
-import { configureStore, combineReducers } from '@reduxjs/toolkit';
+import { combineReducers, legacy_createStore as createStore } from '@reduxjs/toolkit';
+
```
Comment (Gambit142) [inline src/redux/configureStore.js]: - [ ] `createStore` and `combineReducers` should be imported from `'redux'` and not `'@reduxjs/toolkit'`. You can import configureStore from `'@reduxjs/toolkit'` but `createStore` should be imported from `'redux'`. Kindly implement this change. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @ONJoseph 👋, 

Good job so far!

## Highlights
- Installed redux-thunk ✔️
- Created store using thunk middleware ✔️
- All API calls are done with `Axios` ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/redux/configureStore.js (hunk 4 above)
Comment (ONJoseph) [reply src/redux/configureStore.js]: Done

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @ONJoseph, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- You have done a great job working on the requested changes 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**